walkeeper/src/callmemaybe.rs

```diff
@@ -0,0 +1,131 @@
+//!
+//!  Callmemaybe thread is responsible for periodically requesting
+//!  pageserver to initiate wal streaming.
+//!
+use crate::receive_wal::WalStreamInfo;
+use crate::SafeKeeperConf;
+use anyhow::Result;
+use crossbeam_channel::{Receiver, RecvTimeoutError};
+use log::*;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio_postgres::NoTls;
+use zenith_utils::zid::{ZTenantId, ZTimelineId};
+
+async fn request_callback(
+    ps_addr: String,
+    listen_pg_addr: String,
+    pageserver_auth_token: Option<String>,
+    timelineid: ZTimelineId,
+    tenantid: ZTenantId,
+) -> Result<()> {
+    // Connect to the pageserver.
+    let ps_addr_split: Vec<&str> = ps_addr.split(':').collect();
+    let mut config = tokio_postgres::Config::new();
+
+    config
+        .host(ps_addr_split[0])
+        .port(ps_addr_split[1].parse().unwrap());
+
+    if let Some(password) = pageserver_auth_token {
+        config.password(password);
+    }
+
+    debug!("request_callback Connecting to pageserver {}", ps_addr);
+    let (client, connection) = config.connect(NoTls).await?;
+
+    tokio::spawn(async move {
+        if let Err(e) = connection.await {
+            eprintln!("connection error: {}", e);
+        }
+    });
+
+    // Send callmemaybe request
+    let pg_addr_split: Vec<&str> = listen_pg_addr.split(':').collect();
+    let (host, port) = (pg_addr_split[0], pg_addr_split[1]);
+
+    let callme = format!(
+        "callmemaybe {} {} host={} port={} options='-c ztimelineid={} ztenantid={}'",
+        tenantid, timelineid, host, port, timelineid, tenantid
+    );
+
+    let _ = client.simple_query(&callme).await?;
+
+    Ok(())
+}
+
+pub fn thread_main(conf: SafeKeeperConf, rx: Receiver<Arc<WalStreamInfo>>) -> Result<()> {
+    let runtime = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()?;
+
+    let default_timeout = Duration::from_secs(5);
```

Comment:
Might be better to move to a top level constant

Reply:
I'd rather get rid of this entirely and introduce default `recall_period` because I don't see a use case for safekeepers with `recall` disabled.
@petuhovskiy , @arssher , what do you think?

Reply:
Makes sense.